src/lib/server/hyrise_server.cpp

```diff
@@ -18,7 +19,9 @@ void HyriseServer::accept_next_connection() {
 
 void HyriseServer::start_session(boost::system::error_code error) {
   if (!error) {
-    std::make_shared<HyriseSession>(std::move(_socket), _io_service)->start();
+    auto connection = std::make_shared<ClientConnection>(std::move(_socket));
+    // The session will retain a shared_ptr to itself as long as it's alive
+    std::make_shared<HyriseSession>(_io_service, connection)->start();
```

Comment:
What would be a better name for `HyriseServer` and `HyriseSession`?

Reply:
just: Server and ServerSession